src/lib/kdb/kdb_default.c

```diff
@@ -287,7 +287,8 @@ krb5_db_def_fetch_mkey_stash(krb5_context   context,
         goto errout;
     }
 
-    if (!(key->contents = (krb5_octet *)malloc(key->length))) {
+    if (!(key->contents = (krb5_octet *)malloc(
+              sizeof(key->contents[0])*key->length))) {
```

Comment:
My apologies. This fix was provided by a colleague. I rechecked the original Coverity message and it turns out that Coverity was not complaining about the missing multiplication but about the fact that key->length is being read from a file on disk and used without any upper bounds checking.
I had only looked at the fix and from there concluded that the problem was the missing multiplication.